lib/opentsdb_consumer.rb

```diff
@@ -0,0 +1,54 @@
+module OpenTSDBConsumer
+  def self.fetch(host: '', port: '', metric_names: [], metric_options: {})
+    Facade.new(host, port, metric_names).fetch(metric_options)
+  end
+
+  class Facade
+    def initialize(host, port, metric_names)
+      @client       = Client.new(host: host, port: port)
+      @metric_names = metric_names
+    end
+
+    def fetch(metric_options = {})
+      @metric_options = {
+        start: '8h-ago',
+        aggregator: 'sum',
+        downsample: '5m-avg',
+        rate: false,
+        tags: nil,
+      }.merge(metric_options)
+
+      @data ||= query.run(start: metric_options[:start])
+    rescue InvalidTag, InvalidMetric
+      OpenStruct.new datapoints: []
+    end
+
+    private
+
+    attr_reader :client, :metric_names, :metric_options
+
+    def query
+      Query.new(build_metrics, client)
+    end
+
+    def build_metrics
+      metric_names.map do |metric|
+        Metric.new(slice(metric_options, Metric::ATTRIBUTES_WHITELIST).merge(name: metric))
+      end
+    end
+
+    def slice(hash, keys)
```

Comment:
This method does not use the class instance's state, which indicates that it belongs to another class (`Hash` in this case). But I think it would be absolutely ok to just leave it away and solve it with a local variable in the `build_metrics` method. 
Just style...